decred/tests/unit/dcr/test_txscript.py

```diff
@@ -3318,3 +3405,180 @@ def test_get_P2PKH_pCode():
+            pub1.serialize().hex(),
+            pub2.serialize().hex(),
+            pub3.serialize().hex(),
+            pub4.serialize().hex(),
+        )
+    )
+    multisigHash = crypto.hash160(multisig.bytes())
+    multisigP2SH = ByteArray(
+        [opcode.OP_HASH160, opcode.OP_DATA_20, *multisigHash, opcode.OP_EQUAL,]
+    )
+    # fmt: off
+    rawTx = ByteArray([
+        0x01, 0x00, 0x00, 0x00,         # Version
+        0x01,                           # Varint for number of input transactions
+        *bytes(32),                     # Previous output hash
+        0X00, 0X00, 0X00, 0X00,         # Previous output index
+        0x00,                           # Previous output tree
+        0X00, 0X00, 0X00, 0X00,         # Sequence
+        0x01,                           # Varint for number of output transactions
+        0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  # Transaction amount
+        0x00, 0x00,                     # Script version
+        0x01,                           # Varint for length of pk script
+        0x00,                           # Output script
+        0x00, 0x00, 0x00, 0x00,         # Lock time
+        0x00, 0x00, 0x00, 0x00,         # Expiry
+        0x01,                           # Varint for number of input signature
+        0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  # ValueIn
+        0x01, 0x00, 0x00, 0x00,         # BlockHeight
+        0x00, 0x00, 0x00, 0x00,         # BlockIndex
+        # Varint for length of signature script
+        # Signature script
+    ])
+    # fmt: on
+    tx = msgtx.MsgTx.deserialize(rawTx)
+    P2PKHsig = txscript.rawTxInSignature(tx, 0, P2PKH, txscript.SigHashAll, privKey1)
+    multisigSig1 = txscript.addData(
+        txscript.rawTxInSignature(tx, 0, multisig, txscript.SigHashAll, privKey1)
+    )
+    multisigSig2 = txscript.addData(
+        txscript.rawTxInSignature(tx, 0, multisig, txscript.SigHashAll, privKey2)
+    )
+    multisigSig3 = txscript.addData(
+        txscript.rawTxInSignature(tx, 0, multisig, txscript.SigHashAll, privKey3)
+    )
+    emptySig1x = ByteArray(opcode.OP_0)
+    emptySig2x = ByteArray(*([opcode.OP_0] * 2))
+
+    """
+    name (str): Short description of the test.
+    tx (msgtx.Tx): Transaction that pkScript belongs to.
+    idx (int): The output index that pkScript belongs to.
+    pkScript (ByteArray): The script that spends output idx of tx.
+    scriptClass (int): Type of script.
+    addresses list(str): Addresses that comprise a multisig.
+    nRequired (int): Number of addresses required for a multisig.
+    sigScript (ByteArray): The signed script.
+    prevScript (ByteArray): The previous script.
+    want (ByteArray): Merged script.
+    """
+    tests = [
+        dict(
+            name="pay to public key hash",
+            pkScript=P2PKH,
+            scriptClass=txscript.PubKeyHashTy,
+            sigScript=P2PKHsig,
```

Comment:
It doesn't seem to matter much here, but an actual `sigScript` that consisted of only a single signature would have a push_data opcode as prepended by `txscript.addData`. Should we add it?

Reply:
yes